src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3240,20 +3240,20 @@ void SmartScript::ProcessEvent(SmartScriptHolder& e, Unit* unit, uint32 var0, ui
                 if (!creature)
                     return;
 
-                if (!me->IsInRange(creature, 0, (float)e.event.distance.dist))
+                if (!me->IsInRange(creature, 0, static_cast<float>(e.event.distance.dist)))
                     return;
             }
             else if (e.event.distance.entry != 0)
             {
                 std::list<Creature*> list;
-                me->GetCreatureListWithEntryInGrid(list, e.event.distance.entry, (float)e.event.distance.dist);
+                me->GetCreatureListWithEntryInGrid(list, e.event.distance.entry, static_cast<float>(e.event.distance.dist));
 
                 if (!list.empty())
                     creature = list.front();
             }
 
-            if (creature)
-                ProcessTimedAction(e, e.event.distance.repeat, e.event.distance.repeat);
+            if (creature && creature->ToUnit() != nullptr)
```

Comment:
is there a case where creature is != NULL but creature->ToUnit() is NULL ? either the ToUnit() cast is not needed or creature should be declared as Creature* imo